src/UCTNode.cpp

```diff
@@ -473,16 +407,18 @@ void UCTNode::sort_root_children(int color) {
         } else {
             tmp.emplace_back(0.0f, 0, score, child);
         }
-        child = child->m_nextsibling;
     }
 
-    // reverse sort, because list reconstruction is backwards
     std::stable_sort(rbegin(tmp), rend(tmp), NodeComp());
+    std::reverse(begin(tmp), end(tmp));
 
-    m_firstchild = nullptr;
+    // I didn't get the same result with
+    // std::stable_sort(begin(tmp), end(tmp), NodeComp());
+    // maybe because of stable?
 
-    for (auto& sortnode : tmp) {
-        link_child(std::get<3>(sortnode));
+    m_children.clear();
+    for (const auto& sortnode : tmp) {
```

Comment:
Isn't it possible to replace all of this by a single sort call with a custom node comparator now? The deconstruction was done because sorting a linked list is tricky. But now it's a vector.

Reply:
Yes, I've been trying to do the minimum change possible in my pull requests (so that I don't get distracted by fixing random not important things and to make it easier for you to verify the effect of the change).